src/libopensc/card-itacns.c

```diff
@@ -544,6 +489,26 @@ itacns_card_ctl(sc_card_t *card, unsigned long cmd, void *ptr)
 	return SC_ERROR_NOT_SUPPORTED;
 }
 
+static int
+itacns_get_challenge(sc_card_t *card, u8 *rnd, size_t len)
+{
+	size_t max_chunk = 0x100;
+	size_t done = 0;
+
+	if (card->type == SC_CARD_TYPE_ITACNS_CNS_IDEMIA_2021) {
+		max_chunk = 0x20;
+	}
+
+	while (done < len) {
+		size_t req_len = MIN(max_chunk, len - done);
+		int gotlen = default_ops->get_challenge(card, rnd + done, req_len);
+		done += gotlen;
+		LOG_TEST_RET(card->ctx, gotlen, "Get Challenge");
+	}
+
+	return (int) done;
+}
+
```

Comment:
With 410cdf0dcc2dce62f0d4bac00619ddd77920b279 most of the get_challenge functions were rewritten to avoid cycling inside of the function and taking the cycle into the sc_get_challenge keeping the separate driver functions simpler. Can you keep up with this pattern to simplify this function?

Reply:
should be something like this:
```
static int
itacns_get_challenge(sc_card_t *card, u8 *rnd, size_t len)
{
	if (card->type == SC_CARD_TYPE_ITACNS_CNS_IDEMIA_2021 && len > 0x20) {
		len = 0x20;
	}
}
```

Reply:
yes, this looks good